src/lib/utils.ts

```
@@ -67,3 +67,80 @@ export function formatRelativeTime(date: Date): string {
+ * Removes or replaces characters that are invalid in Windows/Mac/Linux filesystems.
+ *
+ * Invalid characters: / \ : * ? " < > |
+ * Also handles control characters and leading/trailing spaces/dots.
+ */
+export function sanitizeFilename(filename: string): string {
+  // Replace invalid characters with underscores
+  let sanitized = filename.replace(/[/\\:*?"<>|]/g, '_');
+
+  // Replace control characters (ASCII 0-31)
+  // eslint-disable-next-line no-control-regex
+  sanitized = sanitized.replace(/[\u0000-\u001f]/g, '');
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)